File: 3-HARD/Python/9-text-justification-dp/proposed-solution/main.py
# Challenge: Given a set of words and a maximum width, implement a dynamic programming algorithm
# to justify text optimally, minimizing penalties for irregular spaces.
# Use object-oriented programming and follow the DRY principle.

from text_justifier import TextJustifier

def main():
    words = ["This", "is", "an", "example", "of", "text", "justification."]
    max_width = 16
    justifier = TextJustifier(words, max_width)
    justified_text = justifier.justify()
    for line in justified_text:
        print(f'"{line}"')

if __name__ == "__main__":
    main()

File: 3-HARD/JavaScript/9-text-justification-dp/proposed-solution/main.js
/*
Challenge:
Given a set of words and a maximum width, implement a dynamic programming algorithm to justify text optimally, minimizing penalties for irregular spaces.

This solution follows DRY principles and is implemented in JavaScript.
*/

import TextJustifier from "./TextJustifier.js";

function main() {
  const words = ["This", "is", "an", "example", "of", "text", "justification."];
  const maxWidth = 16;

  const justifier = new TextJustifier(words, maxWidth);
  const justifiedLines = justifier.justify();

  console.log("Justified Text:");
  justifiedLines.forEach((line) => console.log(`"${line}"`));
}

main();
